src/core/evaluator.js

```diff
@@ -2187,17 +2187,50 @@ class PartialEvaluator {
       spaceInFlowMax: 0,
       trackingSpaceMin: Infinity,
       negativeSpaceMax: -Infinity,
+      notASpace: -Infinity,
       transform: null,
       fontName: null,
       hasEOL: false,
     };
 
+    // Use a cyclic buffer (length === 2) to save the last chars in the
+    // text stream.
+    // It's useful to know when we need to add a whitespace in the
+    // text chunk.
+    const twoLastChars = [" ", " "];
+    let twoLastCharsPos = 0;
```

Comment:
Am I correct in assuming that the reason for using a *cyclic* cache is for overall performance reasons? If so, that's probably worth mentioning in the comment.

The reason for me asking is that this makes the code slightly more complex (when reading it), compared to simply using a combination of e.g. `Array.prototype.push()` and `Array.prototype.shift()` to insert data into the array.

Reply:
Yep, it's a way faster in using this way to do (no need to resize the same array).
I guess there is not so much difference from a user pov, but it's probably better to at least try to minimize the power consumption.
Anyway I updated the comment.

Reply:
Thanks; I figured that we should fry to avoid anyone coming along in the future and thinking that this code could/should just be simplified.